fdbserver/TLogServer.actor.cpp

```diff
@@ -1513,12 +1513,15 @@ std::deque<std::pair<Version, LengthPrefixedStringRef>>& getVersionMessages(Refe
 	return tagData->versionMessages;
 };
 
-Future<Void> waitForMessagesForTag(Reference<LogData> self, TLogPeekRequest const& req) {
+ACTOR Future<Void> waitForMessagesForTag(Reference<LogData> self, TLogPeekRequest const& req) {
 	auto tagData = self->getTagData(req.tag);
 	if (tagData.isValid()) {
 		return Void();
 	}
-	return self->waitingTags[req.tag].getFuture();
+	wait(self->waitingTags[req.tag].getFuture());
```

Comment:
There is a wait() here and the caller is also doing a wait. In addition, there is a call to delay() with seconds = 0. Maybe you can explain the problem in more detail and how these changes solve that problem. Thanks!

Reply:
there is a comment there now. Basically we want to make sure the caller can finish building the right data structure. So when it notifies us, instead of immediately taking up work we call into `wait(delay(0))` to make sure the parent finishes before we get scheduled.

This is more robust. Otherwise changes in the caller might break this in surprising ways.